src/autolabel/models/google.py

```diff
@@ -0,0 +1,180 @@
+import logging
+from functools import cached_property
+from time import time
+from typing import List, Optional
+
+from langchain.schema import Generation, HumanMessage, LLMResult
+from tenacity import before_sleep_log, retry, stop_after_attempt, wait_exponential
+
+from autolabel.cache import BaseCache
+from autolabel.configs import AutolabelConfig
+from autolabel.models import BaseModel
+from autolabel.schema import ErrorType, LabelingError, RefuelLLMResult
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleLLM(BaseModel):
+    SEP_REPLACEMENT_TOKEN = "@@"
+    CHAT_ENGINE_MODELS = ["gemini-pro"]
+
+    DEFAULT_MODEL = "gemini-pro"
+
+    # Reference: https://ai.google.dev/pricing
+    COST_PER_CHARACTER = {
+        "gemini-pro": 0.000125 / 1000,
```

Comment:
let's add other models which might be supported by this api as well. also input and output token prices are different. We have something to handle this in models/openai.py